base_layer/wallet/src/storage/sqlite_db.rs

```diff
@@ -101,11 +265,56 @@ impl WalletBackend for WalletSqliteDatabase {
                 DbKey::CommsSecretKey => {
                     let _ = WalletSettingSql::clear(format!("{}", DbKey::CommsSecretKey), &conn)?;
                 },
+                DbKey::CommsPublicKey => return Err(WalletStorageError::OperationNotSupported),
             },
         }
 
         Ok(None)
     }
+
+    fn apply_encryption(&self, cipher: Aes256Gcm) -> Result<(), WalletStorageError> {
+        let mut current_cipher = acquire_write_lock!(self.cipher);
+        if (*current_cipher).is_some() {
```

Comment:
The `.` operator already does a dereference

```suggestion
        if current_cipher.is_some() {
```

Reply:
orly? TIL